### ISSUE TITLE
change the course drop down list to a input

### ISSUE THREAD

**Issue by XipanXiao** [XipanXiao/1506class#43]
in the schedule editor ('学修安排'), when you create a new one, or edit an existing one, you need to pick a primary (and optionally a secondary) course. Since the course list grows, the drop down gets bigger and bigger.
A nice UI would be an input box with a drop down arrow (refer to 'search_bar.html').
1. If the user is too lazy to type anything, click the drop down arrow should still get the full course list.
2. If something is input, the list is filtered by the input.

Basically this is to add a `datalist` node under the input box, with `options` being all the courses.
The typing/filtering feature is then gained automatically.
